src/pages/admin/create-notes.tsx

```diff
@@ -0,0 +1,418 @@
+    await this.waitForSelection(() => this.state.noteEditObject);
+    const blocksfromHtml = htmlToDraft(this.state.noteEditObject.content);
+    const { contentBlocks, entityMap } = blocksfromHtml;
+    const contentState = ContentState.createFromBlockArray(contentBlocks, entityMap);
+    const date = parse(this.state.noteEditObject.id, "yyyy-MM-dd", new Date())    
+    this.setState({
+      editorState: EditorState.createWithContent(contentState),
+      selectedTags: this.state.noteEditObject.tags,
+      date: date,
+      title: this.state.noteEditObject.title,
+      pdf: this.state.noteEditObject.pdf
+    })
+    this.updateField('title', this.state.noteEditObject.title) // call updateField to set id
+  }
+
+  updateField(field: any, value: any) {
+    let note = this.state.noteObject
+    note[field] = value
+
+    note.id = format(this.state.date, "yyyy-MM-dd")
+    this.setState({ noteObject: note })
+  }
+
+  onChange = (editorState: any) => this.setState({ editorState });
+
+  handlePublishDate = (date: any) => {
+    this.setState({
+      date: date
+    });
+
+    var dayofweek = getDay(date)
+    if (dayofweek !== 0) {
+      this.setState({
+        dateWarning: 'Warning: this date is not a Sunday'
+      })
+    } else {
+      this.setState({
+        dateWarning: ''
+      })
+    }
+  }
+
+  // RENDER FUNCTIONS
+
+  renderEditModal() {
+    return <Modal isOpen={this.state.showEditModal}>
+        <div>Edit existing notes</div>
+        <select onChange={(event:any) => this.setState({ noteEdit: event.target.value})}>
+          <option key="null" value="null">None Selected</option>
+          {this.state.notesList.map((item: any) => {return <option key={item.id} value={item.id}>{item.id}</option>})}
+        </select>
+        <button 
+          onClick={() => this.setState({ 
+            showEditModal: false, 
+            noteEditObject: this.state.notesList.filter((post: any) => post.id === this.state.noteEdit)[0]
+            })
+          }>DONE</button>
+      </Modal>
+  }
+
+  renderMoreOptions() {
+    return (
+      <div>
+
+        <label>
+          Add tags:
+          <input type="text" value={this.state.currentTag} onChange={(event:any) => this.setState({ currentTag: event.target.value})} />
+        </label>
+        <button className="tags-button" onClick={()=>this.setState({selectedTags: this.state.selectedTags.concat(this.state.currentTag), currentTag: ''}) }>Confirm Tag</button>
+        <button className="tags-button" style={{background: "red"}} onClick={() => {this.setState({ selectedTags: [] }); this.updateField('tags', this.state.selectedTags)} }>Clear All Tags</button>
+        <div><b>Current tags (click on tag to delete):</b> {this.state.selectedTags.map((item: any) => <div className="tags-item" onClick={() => {this.setState({selectedTags: this.state.selectedTags.filter((elem: any)=>elem!==item)}); this.updateField('tags', this.state.selectedTags)} }>{item + ", "}</div>)}</div>
```

Comment:
This line is pretty hard to read, probably worth putting it in a separate function.